File: electron/src/api/team.js
import { myrequest } from "./request"

export let api_team_show=(params)=>{
    return myrequest.get('/team/show/',{
    "name":params.name,
    "page":params.page,
    "pagesize":params.pagesize
                })
}

export let api_team_del=(id)=>{
    return myrequest.put('/team/del/',{
    "id":id
                })
}


export let api_team_update=(body)=>{
    return myrequest.put('/team/update/',{
    "id":body.id,
    "name":body.name,
    "phone":body.phone,
    "note":body.note,
    "using":body.using
})
}
export let api_team_add=(body)=>{
    return myrequest.post('/team/add/',{
    "name":body.name,
    "phone":body.phone,
    "note":body.note,
    "using":body.using
})
}

File: LabMangerAPI/LabMangerAPI/Repository/Team.cs

using LabMangerAPI.DTOs;
using LabMangerAPI.Data;
using LabMangerAPI.Models;
using SqlSugar;

namespace LabMangerAPI.Repository;

public class RepositoryTeam:ICrud<Team,Team,RequestTeam.Add,RequestTeam.Show,RequestTeam.Update,RequestTeam.Del>
{
    readonly SqlSugarClient _db=MySqlSugar.Db;
    readonly IUserContext _userContext;
    public RepositoryTeam(IUserContext userContext)
    {   
        _userContext = userContext;
    }

    public async Task<Team> Add(RequestTeam.Add body)
    {
       var result=_db.Insertable(new  Team {
           Name= body.Name,
           Note= body.Note,
           Phone = body.Phone,
       }).ExecuteReturnEntityAsync();
       return await result;
    }
    public async Task<List<Team>> Show(RequestTeam.Show search,RefAsync<int> totalcount,RefAsync<int> totalpage)
    { 
        string? name=search.Name;
        int page=search.Page;
        int pageSize=search.PageSize;
        var exp = Expressionable.Create<Team>();
        exp.And(t=>t.Active==true);
        exp.AndIF(string.IsNullOrEmpty(name)==false,t => t.Name.Contains(name!));
        var result = _db.Queryable<Team>()
            .Where(exp.ToExpression())
            .OrderBy(t => t.Id,OrderByType.Desc)
            .ToPageListAsync(page, pageSize,totalcount, totalpage);

        return await result;
        
    }

    public async Task<Team> Update(RequestTeam.Update body)
    {
        var result = _db.Updateable(new Team {
            Id= body.Id, 
            Name= body.Name,
            Note= body.Note,
            Phone = body.Phone,
        }).ExecuteReturnEntityAsync();
        return await result;
        
    }

    public async Task<bool> Del(RequestTeam.Del body)
    {
        await _db.Updateable<Team>()
            .SetColumns(it => it.Active == false)
            .Where(it => it.Id == body.Id)
            .ExecuteReturnEntityAsync();
        return  true;
    }
}